packages/shared/src/utils/node/ensure-extension-dirs.test.ts

```diff
@@ -0,0 +1,27 @@
+import { ensureExtensionDirs } from '.';
+import { EXTENSION_TYPES } from '../../constants/extensions';
+import { ExtensionType } from '../../types';
+import { dirSync, SynchrounousResult } from 'tmp';
+
+let rootDir: SynchrounousResult;
+
+beforeEach(() => {
+	rootDir = dirSync({ unsafeCleanup: true });
+});
+
+afterEach(() => {
+	rootDir.removeCallback();
+});
+
+describe('ensureExtensionDirs', () => {
+	const types = EXTENSION_TYPES as readonly ExtensionType[];
+	it('returns undefined if the folders exist', async () => {
+		expect(await ensureExtensionDirs(rootDir.name, types)).toBe(undefined);
+	});
+
+	it('throws an error when a folder cant be opened', () => {
+		expect(async () => {
+			await ensureExtensionDirs('/.', types);
+		}).rejects.toThrow(`Extension folder "/interfaces" couldn't be opened`);
```

Comment:
This `/interfaces` name is based on the imported const, but isn't really specific to this particular test/function. we should test with a custom name so we can test the function, rather then test the function combined with the external `types` array (the function should work regardless of input array)

```suggestion
			await ensureExtensionDirs('/.', ['test']);
		}).rejects.toThrow(`Extension folder "/test" couldn't be opened`);
```